code/datums/components/edible.dm

```diff
@@ -0,0 +1,237 @@
+	Components:
+	Drying component (jerky etc)
+	Customizable component (custom pizzas etc)
+	Processable component (Slicing and cooking behavior essentialy, making it go from item A to B when conditions are met.)
+	Dunkable component (Dunking things into reagent containers to absorb a specific amount of reagents)
+
+	Misc:
+	Something for cakes (You can store things inside)
+
+*/
+/datum/component/edible
+	///Amount of reagents taken per bite
+	var/bite_consumption = 2
+	///Amount of bites taken so far
+	var/bitecount = 0
+	///Flags for food
+	var/food_flags = NONE
+	///Bitfield of the types of this food
+	var/foodtypes = NONE
+	///Amount of seconds it takes to eat this food
+	var/eat_time = 30
+	///Defines how much it lowers someones satiety (Need to eat, essentialy)
+	var/junkiness = 0
+	///Message to send when eating
+	var/list/eatverbs
+	///Callback to be ran for when you take a bite of something
+	var/datum/callback/after_eat
+	///Last time we checked for food likes
+	var/last_check_time
+
+/datum/component/edible/Initialize(list/initial_reagents, food_flags = NONE, foodtypes = NONE, volume = 50, eat_time = 30, list/tastes, list/eatverbs = list("bite","chew","nibble","gnaw","gobble","chomp"), bite_consumption = 2, datum/callback/after_eat)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/examine)
+	RegisterSignal(parent, COMSIG_ATOM_ATTACK_ANIMAL, .proc/UseByAnimal)
+	RegisterSignal(parent, COMSIG_ITEM_ATTACK, .proc/UseFromHand)
+	else if(isturf(parent))
+		var/turf/T = parent
+		RegisterSignal(T, COMSIG_ATOM_ATTACK_HAND, .proc/TryToEatTurf)
+
+	src.bite_consumption = bite_consumption
+	src.food_flags = food_flags
+	src.foodtypes = foodtypes
+	src.eat_time = eat_time
+	src.eatverbs = eatverbs
+	src.junkiness = junkiness
+	src.after_eat = after_eat
+
+	var/atom/A = parent
+
+	A.create_reagents(volume, INJECTABLE)
+
+	if(initial_reagents)
+		for(var/rid in initial_reagents)
+			var/amount = initial_reagents[rid]
+			if(tastes && tastes.len && (rid == /datum/reagent/consumable/nutriment || rid == /datum/reagent/consumable/nutriment/vitamin))
+				A.reagents.add_reagent(rid, amount, tastes.Copy())
+			else
+				A.reagents.add_reagent(rid, amount)
+
+/datum/component/edible/proc/examine(datum/source, mob/user, list/examine_list)
+	if(!food_flags & FOOD_IN_CONTAINER)
+		switch (bitecount)
+			if (0)
+				return
+			if(1)
+				examine_list += "[parent] was bitten by someone!"
+			if(2,3)
+				examine_list += "[parent] was bitten [bitecount] times!"
+			else
+				examine_list += "[parent] was bitten multiple times!"
+
+/datum/component/edible/proc/UseFromHand(obj/item/source, mob/living/M, mob/living/user)
+	return TryToEat(M, user)
+
+/datum/component/edible/proc/TryToEatTurf(datum/source, mob/user)
+	return TryToEat(user, user)
+
+///All the checks for the act of eating itself and
+/datum/component/edible/proc/TryToEat(mob/living/eater, mob/living/feeder)
+
+	var/atom/A = parent

```

Comment:
You were using the name `owner` before, why did you rename all usages and switch to this worse name when you could have just added `var/atom/owner = parent` at the top of the procs?